### ISSUE TITLE
BIOM table export

### ISSUE THREAD

**Issue by peterjc** [peterjc/thapbi-pict#520]
For interoperability it might help to offer BIOM format export (v1 uses JSON, or v2 using HDF5), probably most easily done via the format's official Python library https://github.com/biocore/biom-format/

However, one the scale we're currently working with I agree with Robert Edgar's argument on https://www.drive5.com/usearch/manual/otu_table.html that there is value in plain text tables, and the de facto standard seems to be what he calls the QIIME classic OTU table (tabbed text) https://www.drive5.com/usearch/manual/qiime_classic.html

That is referred to in the BIOM code https://github.com/biocore/biom-format/blob/2.1.12/biom/table.py#L4979 as a tab separated (observation x sample) formatted BIOM table, and they support an optional single final column of metadata which could be predicted taxonomy from the classifier.

Matching that TSV output ought to make it trivially easy to convert to BIOM JSON/HDF5 using the biom-format command line tool (or call their Python code internally).

**Comment by peterjc** [comment]
With the current intermediate files (i.e. since v0.11.0 to date, v0.13.x), the information needed for a classic marker ASV x sample style BIOM TSV file is a subset of our reads report. We could provide a simple conversion script, and could go further and call the ``biom`` command line tool to turn it into JSON or HDF5, or use the ``biom`` Python library to do a richer export with metadata.

The _classic_ observation (i.e. ASV) x sample style BIOM TSV only allows for a single per-ASV field like a taxonomic classification. It would not be of limited use for us as an intermediate file. Adopting the BIOM JSON or HDF5 might be more sensible?

**Comment by peterjc** [comment]
The dictionary input to the BIOM table object needs (sequence index, sample index) keys, it will not accept them indexed using sequence or sample identifiers as strings (nor tuples as currently used in our dictionaries):

```pycon
>>> tsv = "rbcL.onebp.tsv"
>>> from thapbi_pict.utils import parse_sample_tsv
>>> from biom import table
>>> seqs, seq_meta, sample_meta, counts = parse_sample_tsv(tsv)
>>> table.dict_to_sparse({(list(seq_meta).index((marker, md5)), list(sample_meta).index(sample)): v for (marker, md5, sample), v in counts.items()}, dtype=int, shape=(len(seq_meta), len(sample_meta)))
>>> table.Table({(list(seq_meta).index((marker, md5)), list(sample_meta).index(sample)): v for (marker, md5, sample), v in counts.items()}, [f"{marker}/{md5}" for (marker, md5) in seq_meta], list(sample_meta))
```

Note this appears to force conversion to float.

**Comment by peterjc** [comment]
Closed via the above pull requests as release v0.14.1
